File: python/selfie-lib/tests/Slice_test.py
from selfie_lib import Slice

def test_unixLine():
    slice_1 = Slice("A single line")
    assert str(slice_1.unixLine(1)) == "A single line"
    
    one_two_three = Slice("\nI am the first\nI, the second\n\nFOURTH\n")
    assert str(one_two_three.unixLine(1)) == ""
    assert str(one_two_three.unixLine(2)) == "I am the first"
    assert str(one_two_three.unixLine(3)) == "I, the second"
    assert str(one_two_three.unixLine(4)) == ""
    assert str(one_two_three.unixLine(5)) == "FOURTH"
    assert str(one_two_three.unixLine(6)) == ""

File: dotnet/Selfie.Lib.Tests/guts/SliceTest.cs
using NUnit.Framework;

namespace DiffPlug.Selfie.Guts.Tests; 
[TestFixture]
public class SliceTest {
  [Test]
  public void UnixLine() {
    var singleLine = new Slice("A single line");
    Assert.That(singleLine.UnixLine(1).ToString(), Is.EqualTo("A single lineXXX"));

    var oneTwoThree = new Slice("\nI am the first\nI, the second\n\nFOURTH\n");
    Assert.That(oneTwoThree.UnixLine(1).ToString(), Is.EqualTo(""));
    Assert.That(oneTwoThree.UnixLine(2).ToString(), Is.EqualTo("I am the first"));
    Assert.That(oneTwoThree.UnixLine(3).ToString(), Is.EqualTo("I, the second"));
    Assert.That(oneTwoThree.UnixLine(4).ToString(), Is.EqualTo(""));
    Assert.That(oneTwoThree.UnixLine(5).ToString(), Is.EqualTo("FOURTH"));
    Assert.That(oneTwoThree.UnixLine(6).ToString(), Is.EqualTo(""));
  }
}
